crates/compiler/solve/src/specialize.rs

```diff
@@ -634,8 +634,17 @@ fn make_specialization_decision<P: Phase>(
                     };
                     match abilities_store.get_implementation(impl_key) {
                         None => {
-                            // Doesn't specialize; an error will already be reported for this.
-                            SpecializeDecision::Drop
+                            match ability_member {
```

Comment:
I know you didn't introduce this (the nesting is my fault), but can we factor out the "abilities_store.get_implementation(impl_key) " switch? This function is super deep and involved.

Reply:
Extract to another function